core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -693,6 +693,10 @@ private[spark] class DAGScheduler(
 
     val jobId = nextJobId.getAndIncrement()
     if (partitions.size == 0) {
+      listenerBus.post(
```

Comment:
Nit: do you want to make sure to specify exactly the same start/end time? it might change between the two lines. This is the only part that gives me pause about this change, but it does seem reasonable to record that it started/ended. Not sure if ... @JoshRosen might have a thought here. The other calls to submit these events were created a long time ago.

Reply:
Still might want to fix this part

Reply:
I don't want to make sure to specify exactly the same start/end time. But when partition is zero, the job will not be scheduled and return immediately.

So, it will be better to post events in `runJob` for semantics.

 ```
     case scala.util.Success(_) =>
        // post a events if partition is zero
       post(jobend)......
        logInfo("Job %d finished: %s, took %f s".format
          (waiter.jobId, callSite.shortForm, (System.nanoTime - start) / 1e9))
      case scala.util.Failure(exception) =>
        logInfo("Job %d failed: %s, took %f s".format
          (waiter.jobId, callSite.shortForm, (System.nanoTime - start) / 1e9))
        // SPARK-8644: Include user stack trace in exceptions coming from DAGScheduler.
        val callerStackTrace = Thread.currentThread().getStackTrace.tail
        exception.setStackTrace(exception.getStackTrace ++ callerStackTrace)
        throw exception
    }
```